viadot/tasks/sharepoint.py

```diff
@@ -0,0 +1,167 @@
+            nrows (int): Number of rows to read at a time.
+            chunks(list): List of data in chunks.
+
+        Returns:
+            List[df]: List of data frames
+        """
+        logging(f"Worksheet: {sheetname}")
+        temp_chunks = chunks
+        i_chunk = 0
+        while True:
+            df_chunk = pd.read_excel(
+                self.path_to_file,
+                sheet_name=sheetname,
+                nrows=nrows,
+                skiprows=skiprows,
+                header=None,
+                **kwargs,
+            )
+            skiprows += nrows
+            # When there is no data, we know we can break out of the loop.
+            if not df_chunk.shape[0]:
+                break
+            else:
+                logging(f" - chunk {i_chunk+1} ({df_chunk.shape[0]} rows)")
+                temp_chunks.append(df_chunk)
+            i_chunk += 1
+        return temp_chunks
+
+    @defaults_from_attrs(
+        "path_to_file",
+        "nrows",
+        "skiprows",
+        "validate_excel_file",
+    )
+    def run(
+        self,
+        path_to_file: str = None,
+        url_to_file: str = None,
+        nrows: int = 50000,
+        skiprows: int = 1,
+        validate_excel_file: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        Run Task ExcelToDF.
+
+        Args:
+            path_to_file (str): Path to Excel file. Defaults to None.
+            url_to_file (str): Link to a file on Sharepoint. Defaults to None.
+            nrows (int, optional): Number of rows to read at a time. Defaults to 50000.
+            skiprows (int): How many rows are we skipping. Defaults to 1.
+            validate_excel_file (bool, optional): Check if columns in separate sheets are the same. Defaults to False.
+
+        Returns:
+            pd.DataFrame: Pandas data frame
+        """
+        self.path_to_file = path_to_file
+        self.url_to_file = url_to_file
+        filename = os.path.basename(self.path_to_file)
+
+        s = Sharepoint()
+        s.download_file(self.url_to_file, filename=filename)
+
+        self.nrows = nrows
+
+        excel = pd.ExcelFile(self.path_to_file)
+        header_to_compare = None
+        chunks = []
+        for sheetname in excel.sheet_names:
+            df_header = pd.read_excel(self.path_to_file, sheet_name=sheetname, nrows=1)
+
+            if validate_excel_file:
+                header_to_compare = self.check_column_names(
+                    df_header, header_to_compare
+                )
+
+            # The first row is the header.
+            chunks = self.split_sheet(sheetname, skiprows, self.nrows, chunks)
+            df_chunks = pd.concat(chunks)
+
+            # Rename the columns to concatenate the chunks with the header.
+            columns = {i: col for i, col in enumerate(df_header.columns.tolist())}
+            df_chunks.rename(columns=columns, inplace=True)
+            df = pd.concat([df_header, df_chunks])
+
```

Comment:
I would add a debug log here showing the shape of the df generated from each sheet, something like `logger.debug(f"Successfully read {df.shape[0]} rows and {df.shape[1]} columns from sheet {sheetname}.")`